### ISSUE TITLE
Adding indent rule to HTML files causes unhandled exception

### ISSUE THREAD

**Issue by bodixon** [angular-eslint/angular-eslint#1092]
**Description and reproduction of the issue**
1. Create new project with `@angular/cli`
```
ng new test-linter
```

2. Add `@angular-eslint/schematics` to your project
```
ng add @angular-eslint/schematics
```
3. Add indent rule to HTML section in `.eslintrc.json`
```
-      "rules": {} 
+      "rules": {
+        "indent": ["warn", 2]
+      }
```
4. Run linter
```
npm run lint
```

AR:
```
An unhandled exception occurred: Cannot read properties of undefined (reading '0')
Occurred while linting /private/tmp/test-linter/test-linter/src/app/app.component.html:2
Rule: "indent"
See "/private/var/folders/0g/hrtcx_nd0qb9w7nsrpr2tkwr0000gn/T/ng-tHIUaW/angular-errors.log" for further details.
```
ER:
Everything works fine

Here is prepared demo project where you can reproduce it:
https://github.com/bodixon/angular-eslint-bug-example

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@angular-eslint/template-parser`  | `14.0.2` |
| `ESLint`                           | `8.19.0` |
| `node`                             | `16.14.2` |

```sh
# `npx ng version`

Angular CLI: 14.0.6
Node: 16.14.2
Package Manager: npm 8.5.0 
OS: darwin x64

Angular: 14.0.6
... animations, cli, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1400.6
@angular-devkit/build-angular   14.0.6
@angular-devkit/core            14.0.6
@angular-devkit/schematics      14.0.6
@schematics/angular             14.0.6
rxjs                            7.5.6
typescript                      4.7.4
```

- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by csisy** [comment]
I have the same issue. Fresh install everything.

**Comment by JamesHenry** [comment]
Hi folks, the `indent` core ESLint rule was written to work on JavaScript code.

You are not able to simply use core ESLint rules on HTML sources, as they are parsed into a completely different data format (called an AST) and the implementation of the rule cannot understand the AST that has been given to it.

You can read our philosophy on using a linter to enforce formatting concerns here: https://github.com/angular-eslint/angular-eslint/blob/main/docs/FORMATTING_RULES.md

The TL;DR is we strongly recommend against it.

If you wanted to have indentation enforced via ESLint in your Angular-flavoured HTML you would need to write a custom rule which did that. You can look at `eslint-plugin-template` within this repo for inspiration on how to write rules for Angular-flavoured HTML.

Hope that helps! As there is no action to be taken for angular-eslint, I am going to close this
